File: ga/glm4_result_2024_6/result/1968E_cpp_glm/27_1968E.cpp
use std::io;
use std::io::{self, Read};

fn solve() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: i32 = input.trim().parse().expect("Please type a number!");

    let t = n;
    println!("1 1");
    println!("2 1");
    let mut t = t - 2;
    if t > 0 {
        println!("{} {}", n, n);
        t -= 1;
    }

    for i in 3..3+t as usize {
        println!("1 {}", i);
    }
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let t: i32 = input.trim().parse().expect("Please type a number!");

    for _ in 0..t {
        solve();
    }
}

File: ga/4omini_result_2024_7/result/1968E_cpp_mini/27_1968E.rs
use std::io::{self, Write};

fn solve(n: usize) {
    let mut output = vec![];
    
    output.push("1 1".to_string());
    output.push("2 1".to_string());
    
    if n > 2 {
        output.push(format!("{} {}", n, n));
    }
    
    let mut t = n.saturating_sub(2);
    
    for i in 3..3 + t {
        output.push(format!("1 {}", i));
    }
    
    // Print all outputs for the current test case
    for line in output {
        println!("{}", line);
    }
}

fn main() {
    let stdin = io::stdin();
    let mut buffer = String::new();
    stdin.read_line(&mut buffer).unwrap();
    
    let t: usize = buffer.trim().parse().unwrap();
    
    for _ in 0..t {
        let mut buffer = String::new();
        stdin.read_line(&mut buffer).unwrap();
        let n: usize = buffer.trim().parse().unwrap();
        
        solve(n);
    }
}